source/Color.cpp

```diff
@@ -130,3 +136,67 @@ Color Color::Multiply(float scalar, const Color &base)
 			scalar * base.color[2],
 			scalar * base.color[3]);
 }
+
+
+
+// Apply color blindness filters to this color.
+Color Color::Filter(Color c)
+{
+	Preferences::ColorFilter filter = Preferences::GetColorFilterMode();
+	// No color filters are enabled, so there's no need to adjust anything.
+	if(filter == Preferences::ColorFilter::NORMAL)
+		return c;
+
+	// Color blindness accessibility filters are enabled.
+	// LMS Daltonization is used to make the colors more distinct for
+	// color-blind players.
+	static float l, m, s, r, g, b;
```

Comment:
Variables should be be defined where they are set for the first time.

```c++
float l = ...;
float m = ...;
// instead of
static float l, m;
// later
l = ...;
m = ...;
```